src/frontends/pytorch/src/op/unfold.cpp

```diff
@@ -0,0 +1,88 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/frontend/pytorch/node_context.hpp"
+#include "openvino/opsets/opset10.hpp"
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace op {
+
+OutputVector translate_unfold(NodeContext& context) {
+    // constants
+    auto const_0 = context.mark_node(opset10::Constant::create(element::i64, Shape{}, {0}));
+    auto const_1 = context.mark_node(opset10::Constant::create(element::i64, Shape{}, {1}));
+    auto const_0_list = context.mark_node(opset10::Constant::create(element::i64, Shape{1}, {0}));
+    auto const_1_list = context.mark_node(opset10::Constant::create(element::i64, Shape{1}, {1}));
+    auto const_neg_1_list = context.mark_node(opset10::Constant::create(element::i64, Shape{1}, {-1}));
+
+    // get inputs and prepare auxiliary nodes
+    auto input = context.get_input(0);
+    auto sizes = context.mark_node(std::make_shared<opset10::ShapeOf>(input));
+    auto input_rank = context.mark_node(std::make_shared<opset10::ShapeOf>(sizes));
```

Comment:
it is great to create PT common helper to compute tensor rank. Please consider it and re-use in other translators:)

Reply:
I think that in general it is not a good idea - we usually use both shape and rank. Calculating rank in an additional sub-procedure will result in redundant shape calculation.